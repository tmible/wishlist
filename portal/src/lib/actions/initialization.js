import { deprive, provide } from '@tmible/wishlist-common/dependency-injector';
import { NetworkService } from './injection-tokens.js';
import * as networkService from './network.service.js';

/**
 * Регистрация зависисмостей для работы с действиями
 * @function initActionsFeature
 * @returns {() => void} Функция освбождения зависисмостей
 */
export const initActionsFeature = () => {
  provide(NetworkService, networkService);
  return () => deprive(NetworkService);
};
