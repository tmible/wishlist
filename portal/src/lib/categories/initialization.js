import { deprive, provide } from '@tmible/wishlist-common/dependency-injector';
import { NetworkService, Store } from './injection-tokens.js';
import * as networkService from './network.service.js';
import { categories } from './store.js';

/**
 * Регистрация зависисмостей и подписка на события для работы с категориями
 * @function initCategoriesFeature
 * @returns {() => void} Функция освбождения зависисмостей и отписки от событий
 */
export const initCategoriesFeature = () => {
  provide(Store, categories);
  provide(NetworkService, networkService);

  return () => {
    deprive(Store);
    deprive(NetworkService);
  };
};
