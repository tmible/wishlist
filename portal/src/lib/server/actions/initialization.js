import { initAddActionStatement } from './statements/add-action.js';

/**
 * Подготовка SQL выражений для работы с действиями
 * @function initActionsFeature
 * @returns {void}
 */
export const initActionsFeature = () => {
  initAddActionStatement();
};
