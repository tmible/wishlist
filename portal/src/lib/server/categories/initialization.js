import { initAddCategoryStatement } from './statements/add-category.js';
import { initDeleteCategoryStatement } from './statements/delete-category.js';
import { initGetCategoriesStatement } from './statements/get-categories.js';
import { initUpdateCategoryStatement } from './statements/update-category.js';

/** @typedef {{ id: number, name: string }} Category */

/**
 * Подготовка SQL выражений для работы с категориями
 * @function initCategoriesFeature
 * @returns {void}
 */
export const initCategoriesFeature = () => {
  initGetCategoriesStatement();
  initAddCategoryStatement();
  initUpdateCategoryStatement();
  initDeleteCategoryStatement();
};
