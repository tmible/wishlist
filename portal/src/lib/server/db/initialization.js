import migrate from '@tmible/wishlist-common/db-migrations';
import { provide } from '@tmible/wishlist-common/dependency-injector';
import { subscribe } from '@tmible/wishlist-common/event-bus';
import Database from 'better-sqlite3';
import { WISHLIST_DB_FILE_PATH, WISHLIST_DB_MIGRATIONS_PATH } from '$env/static/private';
import { RunStatementAuthorized, RunTransaction } from './events.js';
import { Database as DatabaseInjectionToken } from './injection-tokens.js';
import { runStatementAuthorized } from './run-statement-authorized.js';
import { runTransaction } from './run-transaction.js';

/**
 * Создание подключения к БД с отложенным закрытием, миграции БД,
 * регистрация БД в сервисе внедрения зависмостей
 * @function initDB
 * @returns {void}
 */
export const initDB = () => {
  const db = new Database(WISHLIST_DB_FILE_PATH);
  migrate(db, WISHLIST_DB_MIGRATIONS_PATH);
  provide(DatabaseInjectionToken, db);
  process.on('sveltekit:shutdown', () => db.close());

  subscribe(RunTransaction, runTransaction);
  subscribe(RunStatementAuthorized, runStatementAuthorized);
};
