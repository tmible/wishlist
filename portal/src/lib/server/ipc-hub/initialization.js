import { provide } from '@tmible/wishlist-common/dependency-injector';
import { IPCHub, SocketService } from './injection-tokens.js';
import { ipcHub } from './ipc-hub.js';
import * as socketService from './socket.service.js';

/**
 * Регистрация зависимостей для работы с IPC хабом
 * @function initIPCHub
 * @returns {void}
 */
const initIPCHub = () => {
  provide(SocketService, socketService);
  provide(IPCHub, ipcHub);
};

export default initIPCHub;
