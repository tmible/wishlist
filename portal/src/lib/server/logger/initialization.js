import { resolve } from 'node:path';
import { provide } from '@tmible/wishlist-common/dependency-injector';
import { subscribe } from '@tmible/wishlist-common/event-bus';
import pino from 'pino';
import { LOGS_DB_FILE_PATH, LOGS_DB_MIGRATIONS_PATH } from '$env/static/private';
import { createLoggingMiddleware } from './create-logging-middleware.js';
import { CreateLoggingMiddleware } from './events.js';
import { Logger } from './injection-tokens.js';

/**
 * Инициализация логгера. Логи с уровнем INFO и выше пишутся в БД.
 * Логи с уровнем DEBUG и выше пишутся в стандартный вывод.
 * Регистрация логгера в сервисе внедрения зависимостей.
 * Подписка на событие создания промежуточного обработчика
 * @function initLogger
 * @returns {void}
 */
export const initLogger = () => {
  const logger = pino(
    { level: 'debug' },
    pino.transport({
      targets: [
        { target: resolve(import.meta.dirname, 'pino-sqlite-transport.worker.js'), level: 'info' },
        { target: 'pino-pretty', level: 'debug', options: { destination: 1 } },
      ],
      worker: {
        env: {
          LOGS_DB_FILE_PATH,
          LOGS_DB_MIGRATIONS_PATH,
        },
      },
    }),
  );

  provide(Logger, logger);
  subscribe(CreateLoggingMiddleware, createLoggingMiddleware);
};
