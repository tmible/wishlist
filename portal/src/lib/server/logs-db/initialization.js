import migrate from '@tmible/wishlist-common/db-migrations';
import { provide } from '@tmible/wishlist-common/dependency-injector';
import Database from 'better-sqlite3';
import { LOGS_DB_FILE_PATH, LOGS_DB_MIGRATIONS_PATH } from '$env/static/private';
import { Database as DatabaseInjectionToken } from './injection-tokens.js';

/**
 * Создание подключения к БД с логами с отложенным закрытием,
 * миграции БД, регистрация БД в сервисе внедрения зависмостей
 * @function initDB
 * @returns {void}
 */
export const initDB = () => {
  const db = new Database(LOGS_DB_FILE_PATH);
  migrate(db, LOGS_DB_MIGRATIONS_PATH);
  provide(DatabaseInjectionToken, db);
  process.on('sveltekit:shutdown', () => db.close());
};
