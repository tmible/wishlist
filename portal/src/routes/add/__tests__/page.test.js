// @vitest-environment jsdom
import { cleanup, render } from '@testing-library/svelte';
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { initWishlistFeature } from '$lib/wishlist/initialization.js';
import Add from '../+page.svelte';

vi.mock('$lib/components/header.svelte', async () => await import('./mock.svelte'));
vi.mock('$lib/wishlist/components/item-form.svelte', async () => await import('./mock.svelte'));
vi.mock('$lib/wishlist/initialization.js');

describe('add page', () => {
  beforeEach(() => {
    vi.mocked(initWishlistFeature).mockReturnValue(vi.fn());
  });

  afterEach(() => {
    vi.clearAllMocks();
    cleanup();
  });

  it('should be displayed without "wishlist" search param', () => {
    expect(render(Add).baseElement.innerHTML).toMatchSnapshot();
  });

  it('should be displayed with "wishlist" search param', () => {
    vi.spyOn(URLSearchParams.prototype, 'get').mockReturnValueOnce('param');
    expect(render(Add).baseElement.innerHTML).toMatchSnapshot();
  });

  it('should init wishlist feature', () => {
    render(Add);
    expect(vi.mocked(initWishlistFeature)).toHaveBeenCalled();
  });

  it('should destroy wishlist feature', () => {
    const destroyWishlistFeature = vi.fn();
    vi.mocked(initWishlistFeature).mockReturnValueOnce(destroyWishlistFeature);
    render(Add).unmount();
    expect(destroyWishlistFeature).toHaveBeenCalled();
  });
});
