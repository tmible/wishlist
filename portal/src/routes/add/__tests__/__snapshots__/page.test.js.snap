// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`add page > should be displayed with "wishlist" search param 1`] = `"<div><!----><!----><div class="h-full" id="bits-6" dir="ltr" style="position: relative; --bits-scroll-area-corner-height: 0px; --bits-scroll-area-corner-width: 0px;" data-scroll-area-root=""><!----><div class="w-full h-full max-h-dvh" id="bits-7" style="overflow-x: hidden; overflow-y: scroll;" data-scroll-area-viewport=""><div id="bits-8" data-scroll-area-content="" style="min-width: undefined;"><!----><div class="flex flex-col h-full"><!----> <main class="grow-1 pt-6 md:pt-12 pb-0 md:pb-12 md:mx-6 lg:w-1/3 lg:mx-auto"><div class="card h-full md:h-auto bg-base-100 md:shadow-xl"><div class="card-body prose"><!----><!----></div></div></main></div><!----></div></div><!----> <!----><!----><!----><!----> <!----> <!----><!----><!----></div></div>"`;

exports[`add page > should be displayed without "wishlist" search param 1`] = `"<div><!----><!----><div class="h-full" id="bits-1" dir="ltr" style="position: relative; --bits-scroll-area-corner-height: 0px; --bits-scroll-area-corner-width: 0px;" data-scroll-area-root=""><!----><div class="w-full h-full max-h-dvh" id="bits-2" style="overflow-x: hidden; overflow-y: scroll;" data-scroll-area-viewport=""><div id="bits-3" data-scroll-area-content="" style="min-width: undefined;"><!----><div class="flex flex-col h-full"><!----> <main class="grow-1 pt-6 md:pt-12 pb-0 md:pb-12 md:mx-6 lg:w-1/3 lg:mx-auto"><div class="card h-full md:h-auto bg-base-100 md:shadow-xl"><div class="card-body prose"><h3 class="card-title">Что-то пошло не так</h3> <p>Запросите новую ссылку и попробуйте ещё раз</p> <a class="w-full md:flex-1 md:basis-1/3" href="https://t.me/tmible_wishlist_bot"><button class="btn btn-primary w-full">Вернуться к боту</button></a><!----><!----></div></div></main></div><!----></div></div><!----> <!----><!----><!----><!----> <!----> <!----><!----><!----></div></div>"`;
